### PR TITLE
Add annotations to kube-slack deployment

### PR DESCRIPTION
**Pull request by pierluigilenoci**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by PierluigiLenociAkelius** [comment]
There is something missing from me?

**Comment by rimusz** [comment]
@PierluigiLenociAkelius could you please update `readme` with examples of annotations use case?

**Comment by PierluigiLenociAkelius** [comment]
> @PierluigiLenociAkelius could you please update `readme` with examples of annotations use case?

@rimusz the current readme is really poor and contains no examples. It refers to the comments inside the values file. I considered it appropriate not to change this.

**Comment by rimusz** [comment]
> > @PierluigiLenociAkelius could you please update `readme` with examples of annotations use case?
> 
> @rimusz the current readme is really poor and contains no examples. It refers to the comments inside the values file. I considered it appropriate not to change this.

fair enough then :)

**Comment by PierluigiLenociAkelius** [comment]
@mattfarina could you please do something?

**Comment by PierluigiLenociAkelius** [comment]
@viglesiasce could you please verify and approve this?

**Comment by PierluigiLenociAkelius** [comment]
Only your approval is missing.

**Comment by mattfarina** [comment]
cc @jstriebel

File: stable/kube-slack/templates/deployment.yaml [hunk 1]
```diff
@@ -18,6 +18,12 @@ spec:
       labels:
         app: {{ template "kube-slack.name" . }}
         release: {{ .Release.Name }}
+      {{- if .Values.annotations }}
+      annotations:
+      {{- range $key, $value := .Values.annotations }}
+        {{ $key }}: {{ $value | quote }}
+      {{- end }}
+      {{- end }}
```
Comment (mattfarina) [inline stable/kube-slack/templates/deployment.yaml:26]: Can you instead use the `toYaml` function? See https://github.com/helm/charts/blob/bf484e940f3b5abdb8e0d2203e46e4e6628ddec6/stable/spinnaker/templates/ingress/deck.yaml#L6-L7 for an example

Comment (PierluigiLenociAkelius) [reply stable/kube-slack/templates/deployment.yaml]: Done

**Comment by jstriebel** [comment]
@mattfarina Sorry for the late reply, somehow I missed the ping.

Unfortunately the README for kube-slack is outdated since PR #11883. Actually there *is* an example how to use this chart, which does not work since that PR. I did not include the usual table of value-options, as this tends to outdate pretty fast and I personally think commented code is more valuable than an outdated documentation.
I will add a separate PR to fix the README, add some more description about possible env vars and clean-up some details in the chart, thanks for the ping!

@PierluigiLenociAkelius If you think there is more space for improvements in the README, please simply add them.

**Comment by PierluigiLenociAkelius** [comment]
Ops... I think I did something wrong 😢

**Comment by PierluigiLenociAkelius** [comment]
It is absolutely incredible that for such a simple PR we have to wait more than two months. 😞

File: stable/kube-slack/values.yaml [hunk 2]
```diff
@@ -35,3 +35,7 @@ nodeSelector: {}
 tolerations: []
 
 affinity: {}
+
+annotations: {}
+  # Specify which annotations must be added to kube-slack pods
```
Comment (monotek) [inline stable/kube-slack/values.yaml]: Not sure why i was mentioned here.
Anyway, seems there is some issue with the values.yaml, which have to be fixed before pr can be merged.

I guess the whitspace after "pods" should be removed...

**Review by monotek — CHANGES_REQUESTED** [review]

File: stable/kube-slack/values.yaml (hunk 2 above)
Comment (PierluigiLenociAkelius) [reply stable/kube-slack/values.yaml]: 🤦‍♂️

**Review by monotek — APPROVED** [review]

File: stable/kube-slack/templates/deployment.yaml (hunk 1 above)
Comment (jstriebel) [reply stable/kube-slack/templates/deployment.yaml:26]: I think that the indentation spaces before `{{ toYaml .Values.annotations | indent 4 }}` need to be removed, as this is done already by the `indent`. This currently doubles the indentation for the first line.
See [here](https://github.com/helm/charts/blob/master/stable/postgresql/templates/statefulset.yaml#L37) for an example.

Comment (PierluigiLenociAkelius) [reply stable/kube-slack/templates/deployment.yaml]: Done

**Comment by PierluigiLenociAkelius** [comment]
I messed up a lot this PR. I created a new one. Sorry for that.
https://github.com/helm/charts/pull/13312

**Pull request closed without merging**